### ISSUE TITLE
Dependency bounds need tightening

### ISSUE THREAD

**Issue by trilogysci** [hdgarrood/aeson-better-errors#9]
Is there some compatibility issue, which is not specified in cabal file? I am using  the following versions: 
    ghc 7.8.4
    aeson-0.8.0.2
    mtl-2.1.3.1
    transformers-0.3.0.0
    transformers-compat-0.3.3.3
    void-0.7

src/Data/Aeson/BetterErrors/Internal.hs:53:38:
    No instance for (MonadError
                       (ParseError err) (ExceptT (ParseError err) m))
      arising from the 'deriving' clause of a data type declaration
    Possible fix:
      use a standalone 'deriving instance' declaration,
        so you can specify the instance context yourself
    When deriving the instance for (MonadError
                                      (ParseError err) (ParseT err m))

**Comment by trilogysci** [comment]
I think it might be a requirement for mtl >= 2.2 mtl 2.1 does not include an instance for ExceptT

**Comment by hdgarrood** [comment]
Hm, not sure. I did just manage to build it successfully with mtl-2.1.3.1 and transformers-0.3.0.0. Perhaps the instance is coming from transformers-compat? (cabal selected 0.4.something).

**Comment by hdgarrood** [comment]
Yeah, it breaks when I use transformers-compat 0.3.

**Comment by trilogysci** [comment]
I added the following into Internal.hs to get it to work and successfully compile purescript with some additional minor changes. 
```haskell
instance Monad m => MonadError e (ExceptT e m) where
    throwError = ExceptT.throwE
    catchError = ExceptT.catchE
```

**Comment by hdgarrood** [comment]
Fixed by f480fef3fc30398f71df275917a3dfb63a3f834e
